docs/DeveloperGuide.md

```diff
@@ -1,34 +1,845 @@
+
+![Interaction of CalendarCommand and the various major components](images/calendar-diagram.png)
+
+In particular, the diagram below shows the explicit execution flow that `CalendarCommand` takes.
+
+![Explicit execution flow of CalendarCommand](images/addMonthlyCalendar.png)
+
+Given below is an example usage of the `calendar` command. The step by step execution is shown in the sequence diagram above.
+
+**Step 1**  
+The users enters the command `calendar d/05/05/20`. This is captured by the `Ui` component and is subsequently parsed by the `Parser` component that the main component calls.
+
+**Step 2**  
+The `Parser` will construct a `CalendarCommand` object with the `LocalDate` provided by the user input.
+
+> **Note**
+>
+> An `IncorrectCommand` object will be constructed with its specific error message instead according to the error encountered. This can be in the form of no arguments provided or parser fails to parse the date provided.
+
+**Step 3**  
+The `execute()` method in the `CalendarCommand` is then called by the `Atas` class.
+
+The method manages all pre-processing to get the details needed to formulate the calendar. Details include details of tasks that fall within the given month, and the details of the month itself. **The pre-processing work is listed in sequential order below:**  
+
+-   Calibrates an instance of `Calendar` of the `Java.util` class with the provided `LocalDate` and obtain all necessary information about the `Calendar` month.
+
+-   Obtains all `Task` details that falls within the range of the month. This is performed through calling the `getTasksByRange()` method of the `TaskList` class.
+
+-   Duplicates all `RepeatEvent` that is returned from the method above to obtain an `ArrayList` of all `Task` objects that exist within the month.
+
+-   Appends the `Calendar` title and legend to the resultant string that contains the calendar view.
+
+    -   This is done through separate method calls to `addCalendarTitle()` and `addCalendarLegend()` respectively.
+
+-   Appends the main body of the `Calendar` according to the `ArrayList` of `Task` obtained earlier through a method call to `addCalendarBody()`.
+
+-   Constructs a `CommandResult` object with the resultant string that contains the calendar view and returns this object.
+
+> **Note**
+>
+> Since an `Event` can be set to repeat, but is stored within the `TaskList` as a single `Task` object, duplicating a repeat `Event` allows us to obtain the full list of `Tasks` that might occur within the month as separate Task. The decision is further explained in the design considerations subsection.
+
+**Step 4**  
+The `CommandResult` object is subsequently passed to `Ui` component which obtains and prints the `Calendar` view by calling `showToUser()` method of the `Ui` component.
+
+#### 4.6.2. Design Considerations
+
+-   Duplicating `Task` objects instead of keeping the `RepeatEvent` as a single entity like how it is stored in the `TaskList`.
```

Comment:
I am not sure of this design works, but maybe you can use a class level attribute to differentiate between repeat events and task objects